app/models/user.rb

```diff
@@ -32,7 +32,20 @@ def find_for_twitter_oauth(auth)
           image: auth["info"]["image"]
         )
       end
-
+    end
+    def from_omniauth(access_token)
+      data = access_token.info 
+      user = User.where(email: data['email']).first
```

Comment:
Okay so I have just to add a update option for the email changing.

Reply:
@IMPATIENT89  I think we should save the Github User Id instead of their email(same approach is followed in the `find_for_twitter_oauth` method just above this). So regardless of the User changing their emails on their Github account, we will always map to the correct User on our system.
We could use the uid and provider columns for this, if we know for sure we are going to retire Twitter and use Github instead ([post in Slack](https://betacraftio.slack.com/archives/CM741ADRS/p1695614793784709)) - @rtdp

Reply:
@IMPATIENT89 we can't reuse the uid, and provider columns.
As a quick measure, we can simply add github_ prefixed columns to the User table like github_uid to resolve this.

Another approach would be to refactor our existing setup, to allow an User to have multiple modes of logging in.
Perhaps a model named ExternalIdentity (belongs_to :user) with columns :uid, :provider at the minimum and any other columns as necessary.

So Twitter and Github logins will each have a separate row on the ExternalIdentity table.
Adding a Sign in with Google in future would be just adding a row to this table instead of requiring a DB migration.

Let me know what you think of this and which approach you would like to follow.